File: hc-mdm-web/src/main/webapp/resources/app/view/basaccountperiod/BasAccountPeriod.js
/**
 * Description: 会计期间信息表(default)
 * All rights Reserved, Designed ByHc
 * Copyright:   Copyright(C) 2014-2015
 * Company:     Wonhigh.
 * @author:     wugy
 * @date:  2015-03-10 08:50:09
 * @version 1.0.0
 */
Ext.define('Hc_mdm.view.basaccountperiod.BasAccountPeriod', {
	extend: 'Hc_Common.view.BaseSimplePage',
	alias: 'widget.basaccountperiod',
	requires: [
		'Hc_mdm.model.BasAccountPeriod',
		'Hc_mdm.view.basaccountperiod.BasAccountPeriodController',
		'Hc_mdm.view.basaccountperiod.BasAccountPeriodModel'
	],
	controller: 'basaccountperiod',
	viewModel: {
		type: 'basaccountperiod'
	},
	initComponent: function () {
		var me = this;

		Ext.apply(me, {
			searchItems: [
				{xtype: 'textfield', fieldLabel: '会计年月', name: 'accountYearMonth'}
			],

			gridColumns: [
				{dataIndex: 'accountYearMonth', text: '会计年月', editor: true},
				{dataIndex: 'beginDate', text: '开始日期', xtype:'datecolumn',editor:  true},
				{dataIndex: 'endDate', text: '结束日期',xtype:'datecolumn', editor: true},
				{dataIndex: 'orderNo', text: '排列序号', editor: true},
				{dataIndex: 'enableFlag', text: '启用状态(0=禁用 1=启用)', editor: true},
				{dataIndex: 'creator', text: '建档人'},
				{dataIndex: 'createTime', text: '建档时间'},
				{dataIndex: 'modifier', text: '修改人'},
				{dataIndex: 'modifyTime', text: '修改时间'},
				{dataIndex: 'remarks', text: '备注'}
			],
			gridModel: 'Hc_mdm.model.BasAccountPeriod',
			gridPrimaryKey: 'accountYearMonth',
			gridUnionKey: '',
			gridLoadUrl: Hc.mdmPath + 'bas_account_period/list.json',
			gridSaveUrl: Hc.mdmPath + 'bas_account_period/listsave.json',
			gridExportUrl: Hc.mdmPath + 'bas_account_period/do_export.json',
			gridImportUrl: Hc.mdmPath + 'bas_account_period/do_import.json'

		});

		me.callParent();
	}
});

File: hc-mdm-dao/src/main/java/com/hc/scm/mdm/dao/entity/BasAccountPeriod.java
package com.hc.scm.mdm.dao.entity;

import java.util.Date;

/**
 * Description: 请写出类的用途
 * All rights Reserved, Designed Byhc* Copyright:   Copyright(C) 2014-2015
 * Company:     Wonhigh.
 * @author:     wugy
 * @date:  2015-03-10 08:50:09
 * @version 1.0.0
 */
public class BasAccountPeriod {
    /**
     * 会计年月
     */
    private String accountYearMonth;

    /**
     * 开始日期
     */
    private Date beginDate;

    /**
     * 结束日期
     */
    private Date endDate;

    /**
     * 排列序号
     */
    private Integer orderNo;

    /**
     * 启用状态(0=禁用 1=启用)
     */
    private Integer enableFlag;

    /**
     * 建档人
     */
    private String creator;

    /**
     * 建档时间
     */
    private Date createTime;

    /**
     * 修改人
     */
    private String modifier;

    /**
     * 修改时间
     */
    private Date modifyTime;

    /**
     * 备注
     */
    private String remarks;

    /**
     * 
     * {@linkplain #accountYearMonth}
     *
     * @return the value of bas_account_period.account_year_month
     */
    public String getAccountYearMonth() {
        return accountYearMonth;
    }

    /**
     * 
     * {@linkplain #accountYearMonth}
     * @param accountYearMonth the value for bas_account_period.account_year_month
     */
    public void setAccountYearMonth(String accountYearMonth) {
        this.accountYearMonth = accountYearMonth;
    }

    /**
     * 
     * {@linkplain #beginDate}
     *
     * @return the value of bas_account_period.begin_date
     */
    public Date getBeginDate() {
        return beginDate;
    }

    /**
     * 
     * {@linkplain #beginDate}
     * @param beginDate the value for bas_account_period.begin_date
     */
    public void setBeginDate(Date beginDate) {
        this.beginDate = beginDate;
    }

    /**
     * 
     * {@linkplain #endDate}
     *
     * @return the value of bas_account_period.end_date
     */
    public Date getEndDate() {
        return endDate;
    }

    /**
     * 
     * {@linkplain #endDate}
     * @param endDate the value for bas_account_period.end_date
     */
    public void setEndDate(Date endDate) {
        this.endDate = endDate;
    }

    /**
     * 
     * {@linkplain #orderNo}
     *
     * @return the value of bas_account_period.order_no
     */
    public Integer getOrderNo() {
        return orderNo;
    }

    /**
     * 
     * {@linkplain #orderNo}
     * @param orderNo the value for bas_account_period.order_no
     */
    public void setOrderNo(Integer orderNo) {
        this.orderNo = orderNo;
    }

    /**
     * 
     * {@linkplain #enableFlag}
     *
     * @return the value of bas_account_period.enable_flag
     */
    public Integer getEnableFlag() {
        return enableFlag;
    }

    /**
     * 
     * {@linkplain #enableFlag}
     * @param enableFlag the value for bas_account_period.enable_flag
     */
    public void setEnableFlag(Integer enableFlag) {
        this.enableFlag = enableFlag;
    }

    /**
     * 
     * {@linkplain #creator}
     *
     * @return the value of bas_account_period.creator
     */
    public String getCreator() {
        return creator;
    }

    /**
     * 
     * {@linkplain #creator}
     * @param creator the value for bas_account_period.creator
     */
    public void setCreator(String creator) {
        this.creator = creator;
    }

    /**
     * 
     * {@linkplain #createTime}
     *
     * @return the value of bas_account_period.create_time
     */
    public Date getCreateTime() {
        return createTime;
    }

    /**
     * 
     * {@linkplain #createTime}
     * @param createTime the value for bas_account_period.create_time
     */
    public void setCreateTime(Date createTime) {
        this.createTime = createTime;
    }

    /**
     * 
     * {@linkplain #modifier}
     *
     * @return the value of bas_account_period.modifier
     */
    public String getModifier() {
        return modifier;
    }

    /**
     * 
     * {@linkplain #modifier}
     * @param modifier the value for bas_account_period.modifier
     */
    public void setModifier(String modifier) {
        this.modifier = modifier;
    }

    /**
     * 
     * {@linkplain #modifyTime}
     *
     * @return the value of bas_account_period.modify_time
     */
    public Date getModifyTime() {
        return modifyTime;
    }

    /**
     * 
     * {@linkplain #modifyTime}
     * @param modifyTime the value for bas_account_period.modify_time
     */
    public void setModifyTime(Date modifyTime) {
        this.modifyTime = modifyTime;
    }

    /**
     * 
     * {@linkplain #remarks}
     *
     * @return the value of bas_account_period.remarks
     */
    public String getRemarks() {
        return remarks;
    }

    /**
     * 
     * {@linkplain #remarks}
     * @param remarks the value for bas_account_period.remarks
     */
    public void setRemarks(String remarks) {
        this.remarks = remarks;
    }
}